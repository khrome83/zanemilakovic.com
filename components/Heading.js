// @ts-check --jsx

import PropTypes from 'prop-types';

// H1 Element
const H1 = ({ className, children, id }) => (
  <h1 className={`heading ${className}`} id={id}>
    { children }
    <style jsx>{`
      @import './styles/variables.css';

      :root {
        --min: 1.296rem;
        --mid: 2.2265rem;
        --max: 3.157rem;
      }

      .heading {
        font-size: var(--min);
        line-height: calc(var(--min) + var(--heading-height-add));
        margin-top: calc((var(--heading-height-min) - var(--min)) + var(--heading-height-min));
        margin-bottom: var(--heading-height-min);
        position: relative;
        padding: 0;

        @media (--tablet-viewport) {
          font-size: var(--mid);
          line-height: calc(var(--mid) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-mid) - var(--mid)) + var(--heading-height-mid));
          margin-bottom:  var(--heading-height-mid);
        }

        @media (--desktop-viewport) {
          font-size: var(--max);
          line-height: calc(var(--max) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-max) - var(--max)) + var(--heading-height-max));
          margin-bottom:  var(--heading-height-max);
        }
      }
    `}</style>
  </h1>
);

H1.defaultProps = {
  id: null,
  className: null,
};

H1.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  className: PropTypes.string,
};

// H2 Element
const H2 = ({ className, children, id }) => (
  <h2 className={`heading ${className}`} id={id}>
    { children }
    <style jsx>{`
      @import './styles/variables.css';

      :root {
        --min: 1.215rem;
        --mid: 1.9525rem;
        --max: 2.369rem;
      }

      .heading {
        font-size: var(--min);
        line-height: calc(var(--min) + var(--heading-height-add));
        margin-top: calc((var(--heading-height-min) - var(--min)) + var(--heading-height-min));
        margin-bottom: var(--heading-height-min);
        position: relative;
        padding: 0;

        @media (--tablet-viewport) {
          font-size: var(--mid);
          line-height: calc(var(--mid) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-mid) - var(--mid)) + var(--heading-height-mid));
          margin-bottom:  var(--heading-height-mid);
        }

        @media (--desktop-viewport) {
          font-size: var(--max);
          line-height: calc(var(--max) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-max) - var(--max)) + var(--heading-height-max));
          margin-bottom:  var(--heading-height-max);
        }
      }
    `}</style>
  </h2>
);

H2.defaultProps = {
  id: null,
  className: null,
};

H2.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  className: PropTypes.string,
};

// H3 Element
const H3 = ({ className, children, id }) => (
  <h3 className={`heading ${className}`} id={id}>
    { children }
    <style jsx>{`
      @import './styles/variables.css';

      :root {
        --min: 1.138rem;
        --mid: 1.475rem;
        --max: 1.777rem;
      }

      .heading {
        font-size: var(--min);
        line-height: calc(var(--min) + var(--heading-height-add));
        margin-top: calc((var(--heading-height-min) - var(--min)) + var(--heading-height-min));
        margin-bottom: var(--heading-height-min);
        position: relative;
        padding: 0;

        @media (--tablet-viewport) {
          font-size: var(--mid);
          line-height: calc(var(--mid) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-mid) - var(--mid)) + var(--heading-height-mid));
          margin-bottom:  var(--heading-height-mid);
        }

        @media (--desktop-viewport) {
          font-size: var(--max);
          line-height: calc(var(--max) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-max) - var(--max)) + var(--heading-height-max));
          margin-bottom:  var(--heading-height-max);
        }
      }
    `}</style>
  </h3>
);

H3.defaultProps = {
  id: null,
  className: null,
};

H3.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  className: PropTypes.string,
};

// H4 Element
const H4 = ({ className, children, id }) => (
  <h4 className={`heading ${className}`} id={id}>
    { children }
    <style jsx>{`
      @import './styles/variables.css';

      :root {
        --min: 1.067rem;
        --mid: 1.2rem;
        --max: 1.333rem;
      }

      .heading {
        font-size: var(--min);
        line-height: calc(var(--min) + var(--heading-height-add));
        margin-top: calc((var(--heading-height-min) - var(--min)) + var(--heading-height-min));
        margin-bottom: var(--heading-height-min);
        position: relative;
        padding: 0;

        @media (--tablet-viewport) {
          font-size: var(--mid);
          line-height: calc(var(--mid) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-mid) - var(--mid)) + var(--heading-height-mid));
          margin-bottom:  var(--heading-height-mid);
        }

        @media (--desktop-viewport) {
          font-size: var(--max);
          line-height: calc(var(--max) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-max) - var(--max)) + var(--heading-height-max));
          margin-bottom:  var(--heading-height-max);
        }
      }
    `}</style>
  </h4>
);

H4.defaultProps = {
  id: null,
  className: null,
};

H4.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  className: PropTypes.string,
};

// H5 Element
const H5 = ({ className, children, id }) => (
  <h5 className={`heading ${className}`} id={id}>
    { children }
    <style jsx>{`
      @import './styles/variables.css';

      :root {
        --min: 1rem;
        --mid: 1rem;
        --max: 1rem;
      }

      .heading {
        font-size: var(--min);
        line-height: calc(var(--min) + var(--heading-height-add));
        margin-top: calc((var(--heading-height-min) - var(--min)) + var(--heading-height-min));
        margin-bottom: var(--heading-height-min);
        position: relative;
        padding: 0;

        @media (--tablet-viewport) {
          font-size: var(--mid);
          line-height: calc(var(--mid) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-mid) - var(--mid)) + var(--heading-height-mid));
          margin-bottom:  var(--heading-height-mid);
        }

        @media (--desktop-viewport) {
          font-size: var(--max);
          line-height: calc(var(--max) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-max) - var(--max)) + var(--heading-height-max));
          margin-bottom:  var(--heading-height-max);
        }
      }
    `}</style>
  </h5>
);

H5.defaultProps = {
  id: null,
  className: null,
};

H5.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  className: PropTypes.string,
};

// H6 Element
const H6 = ({ className, children, id }) => (
  <h6 className={`heading ${className}`} id={id}>
    { children }
    <style jsx>{`
      @import './styles/variables.css';

      :root {
        --min: 0.937rem;
        --mid: 0.8435rem;
        --max: 0.75rem;
      }

      .heading {
        font-size: var(--min);
        line-height: calc(var(--min) + var(--heading-height-add));
        margin-top: calc((var(--heading-height-min) - var(--min)) + var(--heading-height-min));
        margin-bottom: var(--heading-height-min);
        position: relative;
        padding: 0;

        @media (--tablet-viewport) {
          font-size: var(--mid);
          line-height: calc(var(--mid) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-mid) - var(--mid)) + var(--heading-height-mid));
          margin-bottom:  var(--heading-height-mid);
        }

        @media (--desktop-viewport) {
          font-size: var(--max);
          line-height: calc(var(--max) + var(--heading-height-add));
          margin-top: calc((var(--heading-height-max) - var(--max)) + var(--heading-height-max));
          margin-bottom:  var(--heading-height-max);
        }
      }
    `}</style>
  </h6>
);

H6.defaultProps = {
  id: null,
  className: null,
};

H6.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  className: PropTypes.string,
};

// Heading
const Heading = (props) => {
  const { children, level, ...rest } = props;

  if (level === 1) {
    return <H1 {...rest}>{ children }</H1>;
  } else if (level === 3) {
    return <H3 {...rest}>{ children }</H3>;
  } else if (level === 4) {
    return <H4 {...rest}>{ children }</H4>;
  } else if (level === 5) {
    return <H5 {...rest}>{ children }</H5>;
  } else if (level === 6) {
    return <H6 {...rest}>{ children }</H6>;
  }

  // Defaults to H2
  return <H2 {...rest}>{ children }</H2>;
};

Heading.defaultProps = {
  level: 2,
  id: null,
  className: null,
};

Heading.propTypes = {
  children: PropTypes.node.isRequired,
  level: PropTypes.number,
  id: PropTypes.string,
  className: PropTypes.string,
};

export { Heading as default, H1, H2, H3, H4, H5, H6 };
