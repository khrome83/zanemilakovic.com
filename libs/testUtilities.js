// @ts-check --jsx
/* eslint-disable no-param-reassign */

const cleanSnapshots = (node) => {
  for (const prop in node) {
    if (prop === 'className') {
      const className = node[prop].replace(/jsx-[0-9]+[\s]?/g, '');
      node.className = className || '';
      if (!node.className) {
        delete node.className;
      }
    } else if (typeof node[prop] === 'object') {
      cleanSnapshots(node[prop]);
    }
  }
};

const lib = {
  cleanSnapshots,
};

export { lib as default, cleanSnapshots };
