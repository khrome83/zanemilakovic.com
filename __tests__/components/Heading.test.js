import React from 'react';
import renderer from 'react-test-renderer';
import { cleanSnapshots } from '../../libs/testUtilities';
import Heading, { H1, H2, H3, H4, H5, H6 } from '../../components/Heading';

test('Heading renders Correctly', () => {
  const component = renderer.create(
    <div>
      <Heading level={1} id="foobar" className="large">Heading 1</Heading>
      <Heading level={2} id="foobar" className="large">Heading 2</Heading>
      <Heading level={3} id="foobar" className="large">Heading 3</Heading>
      <Heading level={4} id="foobar" className="large">Heading 4</Heading>
      <Heading level={5} id="foobar" className="large">Heading 5</Heading>
      <Heading level={6} id="foobar" className="large">Heading 6</Heading>
    </div>
  );
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('H1 renders correct', () => {
  const component = renderer.create(<H1 id="foobar" className="large">Heading 1</H1>);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('H2 renders correct', () => {
  const component = renderer.create(<H2 id="foobar" className="large">Heading 2</H2>);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('H3 renders correct', () => {
  const component = renderer.create(<H3 id="foobar" className="large">Heading 3</H3>);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('H4 renders correct', () => {
  const component = renderer.create(<H4 id="foobar" className="large">Heading 4</H4>);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('H5 renders correct', () => {
  const component = renderer.create(<H5 id="foobar" className="large">Heading 5</H5>);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('H6 renders correct', () => {
  const component = renderer.create(<H6 id="foobar" className="large">Heading 6</H6>);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});
