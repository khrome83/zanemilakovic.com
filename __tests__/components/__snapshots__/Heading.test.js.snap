// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 renders correct 1`] = `
<h1
  className="heading large"
  id="foobar"
>
  Heading 1
</h1>
`;

exports[`H2 renders correct 1`] = `
<h2
  className="heading large"
  id="foobar"
>
  Heading 2
</h2>
`;

exports[`H3 renders correct 1`] = `
<h3
  className="heading large"
  id="foobar"
>
  Heading 3
</h3>
`;

exports[`H4 renders correct 1`] = `
<h4
  className="heading large"
  id="foobar"
>
  Heading 4
</h4>
`;

exports[`H5 renders correct 1`] = `
<h5
  className="heading large"
  id="foobar"
>
  Heading 5
</h5>
`;

exports[`H6 renders correct 1`] = `
<h6
  className="heading large"
  id="foobar"
>
  Heading 6
</h6>
`;

exports[`Heading renders Correctly 1`] = `
<div>
  <h1
    className="heading large"
    id="foobar"
  >
    Heading 1
  </h1>
  <h2
    className="heading large"
    id="foobar"
  >
    Heading 2
  </h2>
  <h3
    className="heading large"
    id="foobar"
  >
    Heading 3
  </h3>
  <h4
    className="heading large"
    id="foobar"
  >
    Heading 4
  </h4>
  <h5
    className="heading large"
    id="foobar"
  >
    Heading 5
  </h5>
  <h6
    className="heading large"
    id="foobar"
  >
    Heading 6
  </h6>
</div>
`;
